slick/src/main/scala/slick/jdbc/HsqldbProfile.scala

```diff
@@ -118,8 +121,169 @@ trait HsqldbProfile extends JdbcProfile {
     }
     override val uuidJdbcType = new UUIDJdbcType {
       override def sqlType = java.sql.Types.BINARY
+
       override def sqlTypeName(sym: Option[FieldSymbol]) = "BINARY(16)"
     }
+    override val localTimeType = new LocalTimeJdbcType {
+      override def sqlType = java.sql.Types.TIME
+
+      override def sqlTypeName(sym: Option[FieldSymbol]) = "TIME(3)"
+
+      override def getValue(r: ResultSet, idx: Int): LocalTime = {
+        // Fixes an issue caused because Hsqldb outputs the time in a non-standard format
+        // not adding trailing zeros to the hours. For example: '2:14:41.421' instead of '02:14:41.421'
+        r.getString(idx) match {
+          case null => null
+          case isoTime if isoTime.indexOf(':') == 2 => LocalTime.parse(isoTime)
+          case isoTime => LocalTime.parse(s"0$isoTime")
```

Comment:
This commit implements java.time classes *natively* on all databases when we are allowed to do that. LocalDate, Instant and LocalDateTime are implemented natively on all databases.

If an user wants to store timestamps with timezone in databases with no specific 'TIMESTAMPTZ' implementation, he/her should insert the timestamp using ISO-8601 anyway.

This approach only enforces the user to follow the standard when he/her is working on databases with no specific implementations.

Maybe it would be interesting to add a table in the Slick documentation to explain this.